docs/source/cpp/build_system.rst

```diff
@@ -67,11 +67,55 @@ CMake variables:
 In addition, it will have created some targets that you can link against
 (note these are plain strings, not variables):
 
-* ``arrow_shared`` links to the Arrow shared libraries
-* ``arrow_static`` links to the Arrow static libraries
+* ``Arrow::arrow_shared`` links to the Arrow shared libraries
+* ``Arrow::arrow_static`` links to the Arrow static libraries
+
+For backwards compatibility purposes the ``arrow_shared`` and ``arrow_static``
+targets are also available but we recomend using ``Arrow::arrow_shared`` and
+``Arrow::arrow_static`` respectively.
 
 In most cases, it is recommended to use the Arrow shared libraries.
 
+If Arrow is installed on a custom path instead of a common system one you
+will have to add the path where Arrow is installed to ``CMAKE_PREFIX_PATH``.
```

Comment:
Could you add examples how to add the path to `CMAKE_PREFIX_PATH`?

There are 2 ways to add:

1. `cmake ... -DCMAKE_PREFIX_PATH=...`: CMake variable: https://cmake.org/cmake/help/latest/variable/CMAKE_PREFIX_PATH.html
2. `export CMAKE_PREFIX_PATH=...`: Environment variable: https://cmake.org/cmake/help/latest/envvar/CMAKE_PREFIX_PATH.html

I think the latter is better for most cases because the `CMAKE_PREFIX_PATH` environment variable may be defined in other place.

If we specify `cmake -DCMAKE_PREFIX_PATH=${ARROW_ROOT}`, the pre-defined `CMAKE_PREFIX_PATH` environment variable value is ignored. (I didn't confirm it yet but I think so. Sorry.) We can use the pre-defined `CMAKE_PREFIX_PATH` environment variable value by `cmake -DCMAKE_PREFIX_PATH=${ARROW_ROOT}${CMAKE_PREFIX_PATH:+:${CMAKE_PREFIX_PATH}`. (`${X:+...}` is a variable substitution syntax. See also `${parameter:+word}` in https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html . This is used here to not include an empty path such as `cmake -DCMAKE_PREFIX_PATH=${ARROW_ROOT}:`.)

We can use the variable substitution syntax with `export CMAKE_PREFIX_PATH=...` case:

```bash
export CMAKE_PREFIX_PATH=${ARROW_ROOT}${CMAKE_PREFIX_PATH:+:${CMAKE_PREFIX_PATH}}
```

Reply:
Thanks a lot! I've added a couple of examples, let me know if it makes sense or you think there's something more missing.